pkcs11/src/new/functions/object_management.rs

```diff
@@ -211,4 +126,54 @@ impl Pkcs11 {
             Rv::Error(rv_error) => Err(rv_error.into()),
         }
     }
+
+    // Ignore the unavailable one. One has to call the get_attribute_info method to check which
+    // ones are unavailable.
+    pub fn get_attributes(
+        &self,
+        session: &Session,
+        object: ObjectHandle,
+        attributes: &[AttributeType],
+    ) -> Result<Vec<Attribute>> {
```

Comment:
This could also return a `Result<Vec<Option<Attribute>>` where `None` is returned when the attribute is not available. I figured this would be ok to use as well.

Reply:
A `None` would tell you exactly nothing, though, because it's not a map from `AttributeType` to `Attribute`, so you'd have to somehow check which ones are missing from among the `Some(...)`

Reply:
Or are they presumably in the exact order that you pass them in?

Reply:
Yes they are in the same order so one could zip the `AttributeType` and the returned `Option<Attribute>` iterators to also have the information of what was not available.

Reply:
Which one would you prefer returning? I think I'm ok with any of them, though I prefer returning a `Map<AttributeType, Option<Attribute>>` to just `Vec<Option<Attribute>>`

Reply:
I think I am ok with just returning the `Vec<Attribute>` for now. Callers can use `get_attribute_info` to know if an attribute is available or not